From 3244e9c3e21c11a45df42b62f9c5706425a3e14d Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Wed, 27 May 2020 16:15:15 +0530
Subject: [PATCH 4/5] usdhc-sdslow-changes

This patch will add write-protection pinmux and gpio property for usdhc2

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
index dffe07d0278e..b5894801506a 100755
--- a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
@@ -8,6 +8,7 @@
 
 #include <dt-bindings/usb/pd.h>
 #include "imx8mq.dtsi"
+#define sdslow 1
 
 / {
 	model = "ADLINK LEC i.MX8M";
@@ -771,11 +772,18 @@
 };
 
 &usdhc2 {
+
+#if sdslow
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_usdhc2>;
+#else
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc2>;
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
+#endif
 	cd-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
+        wp-gpios = <&gpio2 20 GPIO_ACTIVE_HIGH>;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
 };
@@ -902,6 +910,9 @@
 
 	pinctrl_reg_usdhc2: regusdhc2grpgpio {
 		fsl,pins = <
+                        MX8MQ_IOMUXC_GPIO1_IO12_GPIO1_IO12              0x41
+                        MX8MQ_IOMUXC_SD2_WP_GPIO2_IO20                  0x41
+                        // SD2_RESET function is SOC SD dedicate, SMARC didn't really connect this GPIO
 			MX8MQ_IOMUXC_SD2_RESET_B_GPIO2_IO19		0x41
 		>;
 	};
-- 
2.17.1

